x-pack/legacy/plugins/lens/public/editor_frame_service/editor_frame/config_panel_wrapper.tsx

```diff
@@ -174,85 +190,329 @@ function LayerPanels(
+                          ...layerDatasourceConfigProps,
+                          core: props.core,
+                          columnId: accessor,
+                          filterOperations: dimension.filterOperations,
+                        }}
+                      />
+                    )}
 
-      <EuiSpacer size="s" />
+                    <EuiButtonIcon
+                      data-test-subj="indexPattern-dimensionPopover-remove"
+                      iconType="cross"
+                      iconSize="s"
+                      size="s"
+                      color="danger"
+                      aria-label={i18n.translate('xpack.lens.indexPattern.removeColumnLabel', {
+                        defaultMessage: 'Remove configuration',
+                      })}
+                      title={i18n.translate('xpack.lens.indexPattern.removeColumnLabel', {
+                        defaultMessage: 'Remove configuration',
+                      })}
+                      onClick={() => {
+                        trackUiEvent('indexpattern_dimension_removed');
+                        props.updateVisualization(
+                          props.activeVisualization.removeDimension({
+                            layerId,
+                            dimensionId: dimension.dimensionId,
+                            columnId: accessor,
+                            prevState: props.visualizationState,
+                          })
+                        );
+                      }}
+                    />
+                  </DragDrop>
+                ))}
+                {dimension.supportsMoreColumns ? (
+                  <DragDrop
+                    className="lnsIndexPatternDimensionPanel"
+                    data-test-subj="indexPattern-dropTarget"
+                    droppable={
+                      dragDropContext.dragging &&
+                      layerDatasource.canHandleDrop({
+                        ...layerDatasourceDropProps,
+                        columnId: newId,
+                        filterOperations: dimension.filterOperations,
+                      })
+                    }
+                    onDrop={droppedItem => {
+                      const dropSuccess = layerDatasource.onDrop({
+                        ...layerDatasourceDropProps,
+                        droppedItem,
+                        columnId: newId,
+                        filterOperations: dimension.filterOperations,
+                      });
+                      if (dropSuccess) {
+                        props.updateVisualization(
+                          activeVisualization.setDimension({
+                            layerId,
+                            dimensionId: dimension.dimensionId,
+                            columnId: newId,
+                            prevState: props.visualizationState,
+                          })
+                        );
+                      }
+                    }}
```

Comment:
For reviewers: Notice that the generated ID is used multiple times in the drop interaction. This guarantees that both the datasource and visualization agree on the ID.